### ISSUE TITLE
Tiered Menu: First sub level doesn't open on hover but all other levels do

### ISSUE THREAD

**Issue by brsanthu** [primefaces/primevue#6069]
### Describe the bug

In Tiered Menu, when we activate the menu, first sub-level menu panel doesn't open when we hover over the item but only when we click on it. After that, all sub-sub-level and their children opens the panel on hover.

Fact that user clicked some button to even show the Tiered menu indicates that all sub-panels after showing the menu should show on hover. This is how all menu systems works as well. This mixed behavior of Primevue is causing lot of confusion.

If we can fix it so once menu is shown, all sub-panels opens on hover or at least make that a config so we can enable it when we render the menu.

### Reproducer

https://stackblitz.com/edit/primevue-3-vite-issue-template-1gf9pm?file=src%2FApp.vue

### PrimeVue version

3.52.0

### Vue version

3.x

### Language

ALL

### Build / Runtime

Vue CLI App

### Browser(s)

Macof Chrome 126.0.6478.127

### Steps to reproduce the behavior

1. In the example click on Show Menu button

### Expected behavior

1. When you click, Level 0 is shown. When you hover over Level 0, Level 2 should have been shown but instead it expects you to click on Level 0. Once you click on Level 0, if you hover over Level 1, Level 2 is shown

**Comment by onekiloparsec** [comment]
I agree, this is an inconsistent behavior. I found a workaround. Note that I am using the `popup` version, with a toggle button (but it should have no impact).

```vue
<template>
...
    <TieredMenu ref="menu" id="overlay_menu" :model="items" popup>
      <template #item="{ item, props, hasSubmenu }">
        /* send both the event and the item to the function below */
        <div @mouseenter="triggerHover($event, item)">
          /* here I have a slot, but it doesn't matter for the issue. */
        </div>
      </template>
    </TieredMenu>
...
</template>

<script setup lang="ts">
  const menu = ref()

  const toggle = (event: any) => {
    menu.value.toggle(event)
  }

  // This is not exposed by Primevue, but it's the minimum you need to be TS correct.
  type ProcessedMenuItem = {
    item: MenuItem
  }

  // This hover special event must actually be done only once.
  const hoverIsDone = ref(false)

  const triggerHover = (event: any, item: any) => {
    if (!hoverIsDone.value) {
      hoverIsDone.value = true
      // One must pass a "processedItem" to the function below. Hence, find it.
      // This is fragile is Primevue modify the name of the computed property "processedItems" but that's how it works today.
      // Match is made with label.
      const processedItem = menu.value.processedItems.find((pi: ProcessedMenuItem) => pi.item.label === item.label)
      // Now trigger the internal click function 
      menu.value.onItemClick({ originalEvent: event, processedItem })
    }
  }
</script>
```

**Comment by bwhitt** [comment]
For reference - I found there is a discussion topic on this you can up vote 
https://github.com/orgs/primefaces/discussions/927

Also would like a prop or some way of controlling primary hover open submenu instead of click. Having a problem with clicking main item vs opening submenu
